chain/epoch_manager/src/lib.rs

```diff
@@ -142,6 +142,63 @@ impl EpochManager {
         Ok(epoch_manager)
     }
 
+    /// Only used in mock network
+    /// Copy the necessary epoch info related to `block_hash` from `source_epoch_manager` to
+    /// the current epoch manager.
+    /// Note that this function doesn't copy info stored in EpochInfoAggregator, so `block_hash` must be
+    /// the last block in an epoch in order for the epoch manager to work properly after this function
+    /// is called
+    #[cfg(feature = "mock_network")]
+    pub fn copy_epoch_info_as_of_block(
+        &mut self,
+        block_hash: &CryptoHash,
+        epoch_id: &EpochId,
+        next_epoch_id: &EpochId,
+        source_epoch_manager: &mut EpochManager,
+    ) -> Result<(), EpochError> {
+        let mut store_update = self.store.store_update();
+        self.save_epoch_info(
+            &mut store_update,
+            epoch_id,
+            source_epoch_manager.get_epoch_info(epoch_id)?.clone(),
+        )?;
+        // save next epoch info too
+        self.save_epoch_info(
+            &mut store_update,
+            next_epoch_id,
+            source_epoch_manager.get_epoch_info(next_epoch_id)?.clone(),
+        )?;
+        // save next next epoch info if the block is the last block
+        if source_epoch_manager.is_next_block_epoch_start(block_hash)? {
+            let next_next_epoch_id =
+                source_epoch_manager.get_next_epoch_id_from_prev_block(block_hash)?;
+            self.save_epoch_info(
+                &mut store_update,
+                &next_next_epoch_id,
+                source_epoch_manager.get_epoch_info(&next_next_epoch_id)?.clone(),
+            )?;
+        }
+
+        let block_info = source_epoch_manager.get_block_info(block_hash)?.clone();
```

Comment:
as we're getting block info from the source_epoch_manager, could we also get the epoch_id and next_epoch_id from there?
(we'd avoid 2 function parameters)